docs/typescript/reference/dbos-class.md

```diff
@@ -0,0 +1,957 @@
+// Run functions directly
+const result1 = await Workflows.processWorkflow("value1");
+const result2 = await workflowsInstance.processInstanceWorkflow("value2");
+```
+
+##### Starting Background Workflows
+Workflow functions can also be started asynchronously using `DBOS.startWorkflow`.  When run with `DBOS.startWorkflow`, the function will run in the background, and a `WorkflowHandle` is returned from `DBOS`.  This [handle](../reference/workflow-handles.md) can be used to [interact with the workflow](#interacting-with-workflows) or await its result with `getResult()`.
+
+```typescript
+DBOS.startWorkflow<T extends ConfiguredInstance>(targetInstance: T): InvokeFunctionsAsyncInst<T>;
+DBOS.startWorkflow<T extends object>(targetClass: T): InvokeFunctionsAsync<T>;
+```
+
+Examples:
+```typescript
+// Run static functions asynchronously and get the result later
+const handle3 = await DBOS.startWorkflow(Workflows).processWorkflow("value3");
+// ...
+const result3 = await handle3.getResult();
+
+// Run instance functions asynchronously and get the result later
+const handle4 = await DBOS.startWorkflow(workflowInstance).processInstanceWorkflow("value4");
+// ...
+const result4 = await handle4.getResult();
+
+```
+
+##### Using Workflow Queues
+By default, the `startWorkflow` method [described above](#starting-background-workflows) always starts the target workflow immediately.  If it is desired to control the concurrency or rate of workflow invocations, a [`queue`](../reference/workflow-queues.md) can be specified.
+
+```typescript
+DBOS.withWorkflowQueue<R>(wfq: string, callback: ()=>Promise<R>) : Promise<R>
+```
+
+The `DBOS.withWorkflowQueue` function runs the provided `callback` function and returns its result.  Any workflows started within `callback` will be run on the queue named by `wfq`.
+
+Example:
+```typescript
+// Create a workflow queue for use below
+const wfq = new WorkflowQueue("example_queue", 10, {limitPerPeriod: 50, periodSec: 30});
+
+// Run a workflow on the queue, and await the result
+const qres = await DBOS.withWorkflowQueue(wfq.name, async ()=>{
+  return await Workflows.processWorkflow("value q");
+});
+
+// Start a workflow on the queue, and return a handle to the running workflow
+const qhandle = await DBOS.withWorkflowQueue(wfq.name, async ()=>{
+  return await DBOS.startWorkflow(Workflows).processWorkflow("value q h");
+});
+//...
+const qhres = await qhandle.getResult();
+```
+
+##### Assigning workflow IDs
+Specifying a workflow ID is useful if the intent is to start a workflow exactly once for a given circumstance.  Assignment of [workflow IDs](../tutorials/idempotency-tutorial.md) can be done with `DBOS.withNextWorkflowID`.
+
+```typescript
+DBOS.withNextWorkflowID<R>(wfid: string, callback: ()=>Promise<R>) : Promise<R>
+```
+
+The `DBOS.withNextWorkflowID` function runs the provided `callback` function and returns its result.  The first workflow started within `callback` will be assigned the ID specified by `wfid`.
```

Comment:
The callback function currently allows running multiple workflows with the same workflow ID(?) Can we have a clear error message to disallow this?